src/optimal_retention.rs

```diff
@@ -206,251 +161,160 @@ pub fn simulate(
 
     let mut rng = StdRng::seed_from_u64(seed.unwrap_or(42));
 
-    let mut init_ratings = Array1::zeros(deck_size);
-    init_ratings.iter_mut().for_each(|rating| {
-        *rating = first_rating_choices[first_rating_dist.sample(&mut rng)];
-    });
+    let mut cards = Vec::with_capacity(deck_size);
 
-    // Main simulation loop
-    for today in 0..learn_span {
-        let old_stability = card_table.slice(s![Column::Stability, ..]);
-        let has_learned = old_stability.mapv(|x| x > 1e-9);
-        let old_last_date = card_table.slice(s![Column::LastDate, ..]);
+    if let Some(existing_cards) = existing_cards {
+        if existing_cards.len() > deck_size {
+            return Err(FSRSError::InvalidDeckSize);
+        }
+        cards.extend(
+            existing_cards
+                .into_iter()
+                .filter(|card| card.stability > 1e-9),
+        );
+    }
 
-        // Updating delta_t for 'has_learned' cards
-        let mut delta_t = Array1::zeros(deck_size); // Create an array of the same length for delta_t
+    if learn_limit > 0 {
+        let init_ratings = (0..(deck_size - cards.len())).map(|i| {
+            // Initialize stability and difficulty for new cards
+            let rating = first_rating_choices[first_rating_dist.sample(&mut rng)];
+            let offset = first_rating_offsets[rating - 1];
+            let stability =
+                stability_short_term(w, w[rating - 1], offset, first_session_lens[rating - 1]);
+            let day = i / learn_limit;
+
+            let ivl = next_interval(stability, desired_retention);
+            let retrievability = power_forgetting_curve(ivl, stability);
+
+            let upper = min(day, learn_span);
+            for i in 0..upper {
+                memorized_cnt_per_day[i] += retrievability;
+            }
 
-        // Calculate delta_t for entries where has_learned is true
-        izip!(&mut delta_t, &old_last_date, &has_learned)
-            .filter(|(.., &has_learned_flag)| has_learned_flag)
-            .for_each(|(delta_t, &last_date, ..)| {
-                *delta_t = today as f32 - last_date;
-            });
+            if day < learn_span {
+                learn_cnt_per_day[day] += 1;
+                cost_per_day[day] += learn_costs[rating - 1];
+            }
```

Comment:
In previous framework, the new/review order is "show after reviews". Is it possible to implement that order in current framework?